rfcs/web_extensions.md

```diff
@@ -0,0 +1,119 @@
+# RFC #219: Add support for WebExtensions in WPT
+
+## Summary
+
+The WebExtensions API extends the capabilities of the browser. Adding support
+for WebExtensions in the web-platform-tests will increase interoperability
+and will help drive the standardization of this API.
+
+This RFC proposes adding a new `testharness.js` wrapper type, `.extension.js`, to handle
+testing this API, in addition to using `testdriver.js` to load and unload extensions.
+   
+##  Details
+  
+Using `testdriver.js`, we added support for testing the WebExtensions API by loading
+a web extension designed to test the functionality of a specific API.
+
+These tests won't leverage `testharness.js` directly since most of the test execution
+is handled within the extension, via the `browser.test` API. We’ve elected to use
+this API since all participating browser vendors use `browser.test` internally and
+can easily port over existing tests to the web-platform-tests.  
+
+The `browser.test` API will also be used to send messages back to the web page to notify the testharness
+when a test has started and finished. For these cases, it is up to the user agent to send the message
+"test-started" and "test-finished", respectively. In addition, when recording the result for 
+a `browser.test.assert...()` method, it is up to the user agent to send the message "assert"
+or "assert-equality". 
+
+Because these tests don’t leverage `testharness.js` directly, we’ve introduced a new
+wrapper type, `.extension.js`, that will create the necessary boilerplate to
+convert these messages into the corresponding assertions in the `testharness.js`.
+   
+A proposed patch is available at https://github.com/web-platform-tests/wpt/pull/50648.
```

Comment:
Is the PR in sync with the RFC? I wonder if RFC should also discuss how the runner would be resolving path to the extension. It appears that one option is to resolve extension path relative to the test path and another option is to require tests to provide path relative to the WPT root folder?

Reply:
In the PR, I have it set to get the extension path relative to the test path [here](https://github.com/web-platform-tests/wpt/pull/50648#discussion_r1957778696). i can add a comment in the RFC detailing that